torchrl/envs/gym_like.py

```diff
@@ -39,11 +57,30 @@ class default_info_dict_reader:
 
     """
 
-    def __init__(self, keys=None):
+    def __init__(
+        self,
+        keys: List[str] = None,
+        spec: Union[Sequence[TensorSpec], Dict[str, TensorSpec]] = None,
+    ):
         if keys is None:
             keys = []
         self.keys = keys
 
+        if isinstance(spec, Sequence):
+            if len(spec) != len(self.keys):
+                raise ValueError(
+                    "If specifying specs for info keys with a sequence, the "
+                    "length of the sequence must match the number of keys"
+                )
+            self._info_spec = dict(zip(self.keys, spec))
+        else:
+            if spec is None:
+                spec = {}
+
+            self._info_spec = {
+                key: spec.get(key, UnboundedContinuousTensorSpec()) for key in self.keys
```

Comment:
MyPy gives me a warning here because `TensorSpec` has abstract method `index` which `UnboundedContinuousTensorSpec` does not define.